### ISSUE TITLE
order.put_market, amount, inconsistency 

### ISSUE THREAD

**Issue by AndrewSamokhvalov** [viabtc/viabtc_exchange_server#32]
Hey @haipome ! We have noticed inconsistency in what we have to specify in the field `amount` for `order.put_limit` and `order_put_market`, and I am wondering is that was done intentionally or not.

In the command `order.put_limit` `amount` corresponds to the number of stocks user wants to `buy/sell`.

In the command `order.put_market` when we use `bid` side `amount` corresponds to number of money user wants to spend on buying the stock, and when we use `ask` side `amount` corresponds to the stock user want to sell. Wouldn't it be better to make it consistent or probably I am missing something? Thanks!

| Command  | Side | Meaning of `amount`  |
| ------------- | ------------- | ------------- |
| order.put_limit  |  ask | stock  |
| order.put_limit  |  bid | stock  |
| order.put_market  |  ask | stock  |
| order.put_market  |  bid | money  |

**Comment by haipome** [comment]
Yes. When use put_market, the amount filed meaning is different in ask and bid. When you want use put_market to sell, the amount means the stock amount want to sell. when you want to buy, the amount means the money amount want to buy, because the price is not float.

**Comment by esseswann** [comment]
We also have a problem with consistency of `order.put_market`, for such a use-case:
- User intends to make a Bid order for specific amount of stock
- To maker an order or to show user an estimation we have to calculate amount of money
- Market can change during calculation and the resulting amount of stock will differ from what user intended
This puts us in a position where for Bid market order we don't allow user to input stock while for Ask market order to input money.
Consistency would at least allow us to always expect input in stock like it's done on Bitfinex

**Comment by haipome** [comment]
@esseswann this advice is good. I can add stock or money amount in market order as an optional.

**Comment by esseswann** [comment]
Thanx for response and agreeing to add this feature. Can you please tell if you plan to do this any time soon so we could correct our estimates?

**Comment by xtuyaowu** [comment]
建了一个交易所撮合交流群： 862266237
